src/mon/OSDMonitor.cc

```diff
@@ -4872,6 +4900,34 @@ void OSDMonitor::clear_pool_flags(int64_t pool_id, uint64_t flags)
   pool->unset_flag(flags);
 }
 
+string OSDMonitor::make_snap_epoch_key(int64_t pool, epoch_t epoch)
+{
+  char k[40];
```

Comment:
rocksdb and leveldb still do prefix encoding, so the full key isn't stored for every item--only the suffix that changes.  i'm not worried about the size (or even performance) here since we aren't doing many queries over this data (it's just the items that are in the process of being purged)... i'm more worried about code and data schema clarify.

Reply:
(fixed the buffer sizes tho!)